readthedocs/doc_builder/backends/sphinx.py

```diff
@@ -272,34 +249,29 @@ def get_config_params(self):
 
     def append_conf(self):
         """
-        Find or create a ``conf.py`` and appends default content.
+        Find a ``conf.py`` and appends default content.
 
         The default content is rendered from ``doc_builder/conf.py.tmpl``.
         """
+        if self.config_file is None:
+            raise ProjectConfigurationError(ProjectConfigurationError.NOT_FOUND)
 
-        # Generate a `conf.py` from a template
-        #
-        # TODO: we should remove this feature at some point to move forward
-        # with the idea of remove magic from the builders.
-        if not self.config_file:
-            self._write_config()
+        self.config_file = self.config_file or self.project.conf_file(self.version.slug)
 
         try:
-            self.config_file = (
-                self.config_file or self.project.conf_file(self.version.slug)
-            )
+            if not os.path.exists(self.config_file):
+                raise UserFileNotFound
+
             # Allow symlinks, but only the ones that resolve inside the base directory.
             outfile = safe_open(
                 self.config_file, "a", allow_symlinks=True, base_path=self.project_path
             )
             if not outfile:
```

Comment:
the result from `safe_open` is always a context manager or an exception, so we can omit this check.